colcon_ros/package_identification/ignore.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2016-2018 Dirk Thomas
+# Licensed under the Apache License, Version 2.0
+
+from colcon_core.package_identification \
+    import PackageIdentificationExtensionPoint
+from colcon_core.package_identification import SkipLocationException
+from colcon_core.plugin_system import satisfies_version
+
+
+class IgnorePackageIdentification(PackageIdentificationExtensionPoint):
+    """
+    Ignore paths containing a `_order_isolated_packages.py` file.
+
+    Workspaces built with `ament_tools` don't contain any IGNORE marker files
+    in the install space.
+    To avoid packages being found in that directory by the discovery
+    extensions using default values it is being ignored by this extension.
+    """
+
+    # the priority needs to be higher than all other extensions
+    PRIORITY = 1000
+
+    def __init__(self):  # noqa: D107
+        super().__init__()
+        satisfies_version(
+            PackageIdentificationExtensionPoint.EXTENSION_POINT_VERSION,
+            '^1.0')
+
+    def identify(self, desc):  # noqa: D102
+        ignore = desc.path / '_order_isolated_packages.py'
```

Comment:
would it make sense to have this defined globally as a list / dict that can be extended for artifacts generated by other build tools ? (catkin drops a `_setup_util` I believe, maybe catkin tools does something else ?)

Reply:
I updated the patch to also consider `_setup_util.py` files. I am not sure how a globally defined list would be beneficial and how it would be extended. Therefore I kept it simple.

Reply:
That's fine, now we have a list with comments explaining what build tool creates them so we can point users to that if they want to contribute a custom "ignore rule"